src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1468,64 +1468,87 @@ internal static decimal ParseDecimal(ReadOnlySpan<char> value, NumberStyles styl
 
         private static unsafe bool NumberBufferToDecimal(ref NumberBuffer number, ref decimal value)
```

Comment:
As a general suggestion, this function is using the canonical `Try` semantics (e.g. `TryParse()`, `TryAdd()`, etc). Can we update the name to be explicit about that (i.e. `TryNumberBufferToDecimal()`)?

Reply:
If you are going to address this, make the same fix for `NumberBufferToDouble` and also change `value` to `out`.

Reply:
Using `out` instead of `ref` here would be slower. It's a private function that does not exactly conform to the canonical Try semantics.